File: LibrarySearcEnginePython/Video.py
from Media import Media


class Video(Media):

    def __init__(self, callnumber, title, subject, description, distributor, notes, series, label):
        super(Video, self).__init__(callnumber, title, subject, notes)
        self.description = description
        self.distributor = distributor
        self.series = series
        self.label = label

    def searchother(self, other):
        return other in self.description or other in self.distributor

    def display(self):
        print("--------------------VIDEO-------------------")
        super(Video, self).display()
        print(" Description: " + self.description + "\n" +
              " Distributor: " + self.distributor + "\n" +
              " Series: " + self.series + "\n" +
              " Label: " + self.label + "\n")

File: Library_search_engine/video.cpp
/*

FileName: video.cpp
Description: stores all the video values except the once that media holds, has a display
function and search other function.
Author: Akshay Joshi

*/

#include <iostream>
#include <iomanip>
#include "video.h"
#include <string>
using namespace std;

	Video::Video(const string& video_CN,const string& video_T,const string& video_subj, const string& video_des,const string& video_distro,const string& video_notes, const string& video_series,const string& video_labels)
:Media(video_CN, video_T, video_subj,video_notes), description(video_des), distributor(video_distro), series(video_series), labels(video_labels)
{}
void Video::Display()const
{
	cout << "--------------------VIDEO-------------------" << endl;
	Media::Display();
	cout << "Description:  "<<description << endl;
	cout << "Distributor:  "<<distributor << endl;
	cout << "Series:  "<<series << endl;
	cout << "Labels:  " << labels << endl;
	cout << "--------------------------------------------\n" << endl;
}
bool Video::SearchOther(const string& keywords)const
{
	bool flag;
	{
		if((this->description.find(keywords)!= string::npos) || (this->distributor.find(keywords)!= string::npos))
		{
			flag= true;
		}
		else
		{
			flag= false;
		}
	}
	bool mflag=Media::SearchOther(keywords);
	if (flag || mflag)
	{
		return true;
	}
	else{
		return false;
	}
}
